### DISCUSSION TITLE
Dueling bots outside Orgrimmar, Stormwind etc

### DISCUSSION THREAD

**Discussion by Rozza122** [mod-playerbots/mod-playerbots#455] [Ideas]
One thing i would love to see is dueling bots outside the main city's with a config option for the amount of bots to spawn and duel, i think it would add a lot to the immersion.

i wouldn't mind trying to do this myself to get to grips with PlayerBot development (have been coding for 15 years, just not in c++) anyone is welcome to take on as no doubt you will do a better job than i would but any tips would be appreciated on roughly where the changes would need to be.

**Comment by Dreathean** [comment]
This is a pretty cool idea

**Comment by Rozza122** [comment]
Well, managed to get them to spawn outside Stormwind and Orgrimmar. getting there (alot left to work out though, much respect to the current devs of Playerbots!)
![sc0](https://github.com/user-attachments/assets/20b8a088-7e11-48f4-b028-154b920ccb19)
![sc1](https://github.com/user-attachments/assets/252e2f27-59d7-4a01-802f-528a1e7f0a34)

**Comment by Rozza122** [comment]
can anyone advise on the method to set the bots to duel / invite a player or bot to a duel, i can see there is a nc "duel" value and a rpg "start duel" but have not managed to get a bot to duel / invite to duel yet.

im convinced the strat "start duel" is bugged, i have added to bots but it doesn't seem to be triggered at any point.

**Comment by Rozza122** [comment]
am starting the idea in lua (find it less annoying to debug and prototype in but will change to c++ once i have it figured out in lua) here is what I'm hoping to put into playerbots at some point (although might be better as a module?) , it is essentially a way to to teleport bots to areas if within a certain level if there is a real player in the area (also for pvp areas in outland etc) here is an example of level 1-80 bots (upto to 600 bots, extreme example):

https://www.youtube.com/watch?v=iMeqL20ySn0

and then a more sensible upto 40 (level 80 only):

https://www.youtube.com/watch?v=FqgG_LlHVh0

**Reply by ThePenguinMan96** [reply]
This is awesome! Bots that duel outside of org/stormwind/dalaran sewers is a dream of mine. I would love to see this in action. I just implemented PVP specs as well

**Reply by Rozza122** [reply]
Good timing with the new specs :), i have uploaded the script here https://github.com/Rozza122/Player-Bots-Pvp-Teleporter its not anywhere near finished but you are welcome to try it out.

edit, re wrote 99% will push up soon.

**Reply by mad0** [reply]
> Good timing with the new specs :), i have uploaded the script here https://github.com/Rozza122/Player-Bots-Pvp-Teleporter its not anywhere near finished but you are welcome to try it out.
> 
> edit, re wrote 99% will push up soon.

Any update?

**Comment by hilch1981-prog** [comment]
Great work on this! I've been envious of the EmuCoach MoP version, so it's awesome to see this being developed for the community. Can't wait for the final release!
